### ISSUE TITLE
Aileron / Ail=> Flaps mixer needs adjusting

### ISSUE THREAD

**Issue by Jedsters** [FrSkyRC/ETHOS-Feedback-Community#2146]
A common scenario (at least for gliders), is to have a 4 servo wing, 2 for aileron and 2 for flaps. 
It is also common to have 'dual rates' on the ailerons which can be done via the Weights / Rates pointing to a Var mixer. 
Then we commonly want to use Flaps as Ailerons which can be done with the Ail=> Flaps mixer. But of course, it is common that the flaps don't move as much as the aileron, maybe they only move 50% of the aileron movement. 

However, now we have a problem. When using dual rates on the ailerons, we obviously want the flaps movement to be reduced too  - otherwise the flaps could end up moving more than the ailerons! So, maybe we use the same Var mixer on the Ail=> Flaps weight. 
But now we have a problem as we have no easy way to reduce the flap throw as a percentage of aileron throw. We now have to add more Mixers, channels etc..

There are a couple of possible solutions here. 
1) Why not just use the Ailerons mixer and add a Channel Weight (maybe only to 2 of the channels, but allowing a Var Mixer as a source). 
2) Use the existing Ail=> Flaps mixer and add an additional Weight, e.g. Percentage of Aileron Weight, so that both dual rates and Percentage of aileron weight would get applied. 

I've no doubt there are other options, but these seem the simplest. 

We need to ensure that the same dual rates are applied to both ailerons and flaps, that the same expo is applied to both, that the same differential is applied to both ailerons and flaps, that the flaps can be set to a percentage of the aileron movement.

I notice that some of the mixers have 'Weight' while others have 'Channel Weight'. There seems to be a use case for a mixer to have both or at least multiple weights for some channels.

Thanks

**Comment by Jedsters** [comment]
Whether 1 or 2 is implemented possibly depends on whether it is desired that Aileron trim should affect the Flaps or not. 
If Aileron Trim should affect flaps then you'd need to go down the route or 1), or if flaps should not be affected by Aileron trim then route 2 would be needed.

**Comment by mawzthefinn** [comment]
You can do this, but it's a bit of a workaround.

Set your base percentage with a curve, then use the Weights called from a VAR mixer to mirror Aileron rates.

That said, why would you want full-span ailerons with low aileron rates? My setups all only engage full-span ailerons when ailerons are max rate, because when you are in lower rates, you don't gain anything with full-span ailerons, that feature is generally used to increase roll rate when you are already maxed out for aileron rates.

Full-span ailerons effectively increases aileron rates. 

The extra tuneability of this mix would be nice, but I'd like to understand why you would be running full-span and low rates at the same time.

**Comment by Jedsters** [comment]
Yes, I have a work-around thanks, although it's not ideal. I'm not a big fan (if I understand you) of using a curve as a weight, it's quite fiddly to adjust as a var, even if it isn't adjusted much after initial setup. A shame you can't use a Var as input to a curve weight.
Are you suggesting e.g. aileron only for 0 - 80% ail stick travel and then include the 100% flap travel in the last bit of aileron travel (via a curve no doubt)? Interested to learn how you set yours up.
I've spoken with a number of people over the years and most I've spoken to do similar to myself - but that's a relatively small sample. 
The first scenario is on some gliders which have a little dihedral where the flap becomes the aileron, if you try and get them going up by the same amount, the sides of the ail and flap can collide.
How to explain the other reason why... I'll try! Efficiency is the theory. As you say, the aileron has more effect than a flaperon as it is further out so more roll 'torque'. If you move the flap up / down as much as the aileron yes you'll get a bit more roll torque although nothing like what you get from the aileron, but you get the same drag from the flap as you do from the aileron. So disproportionally more drag, a bit more roll torque for a lot more drag. Doing e.g. half flaperon is a balance between more roll torque without too much more drag. I guess in theory you'd want a full width aileron which varies a bit like a prop blade does (I'm talking just about the general shape shape, it doesn't work as an analogy beyond that), balancing roll torque against drag across the width. 
I take the point about not wanting to use flaperon until you've maxed out aileron, but I don't think that nullifies the above, even if it is done after that. Not sure whether full aileron only is better than 2/3 ail, 1/3 flaperon as a balance between roll torque and drag. 
Hope you understand even if you don't agree.

**Comment by mawzthefinn** [comment]
The curve is definitely a workaround, because it's a pain to edit as you note.

What I'm suggesting is to have the Aileron-> flap mix disabled when you aren't in full rates. I personally only activate it in Speed mode on most of my setups, but I tend to loaf around in Cruise most of the time. 

The flaps actually add fairly little roll input unless you're banging the sticks, so the drag difference for smaller inputs is quite small and there's not that much drag added from the input (the nose wagging from adverse yaw in a sailplane is more because of the length of the wing giving more moment than actual drag increase). So adding the full-span mix has minimal improvement on the drag of the input unless you are using larger deflections, where you want it anyways to improve the roll rate. The best solution for minimizing drag from roll inputs is actually using wide-chord ailerons, as most competition gliders do.

This is also why strip ailerons on power models mostly suck.

**Comment by Jedsters** [comment]
To my mind, if you're going slow e.g. in acro mode you may want full travel flaperons as the drag isn't important, however in speed mode you want to minimise the drag so half flaperons are perhaps more appropriate. 
What I would mention is that Mike Shellim's excellent templates also allow for flaperons as a percentage of ailerons. Again, it's all do-able without my request, it's just messier. I'm far from alone.

**Comment by Jedsters** [comment]
Worthy of consideration? 
Ail->Flap needs all the same input as Ail (i.e. Expo curve, Dual rates, Differential) plus an extra weight to reduce the amount of flaperon to be a percentage of aileron.
Thanks

**Comment by Jedsters** [comment]
Bump

**Comment by Toffs** [comment]
Hi Jedsters,
Yes, I came across this very issue a little while back. Every point you’ve marked down is exactly what I also was seeking.
However, I found that using Vars to achieve similar requirements was more useful since I could apply FlightMode specific settings for each mix value.
EG, changing Flaperon % or Differential depending on what FM was selected.

Perhaps my requirement (and solution using Vars) is too complicated for the pre-set mix use case, and your excellent suggestion above is better for that.

**Comment by Jedsters** [comment]
Bump again...
